### DISCUSSION TITLE
DNS not working once kernel is booted

### DISCUSSION THREAD

**Discussion by icsy7867** [ipxe/ipxe#988] [Q&A]
Not directly an IPXE issue, but I am hoping somewhere here has some helpful experience.  It took me forever to figure out why I couldnt boot to live environments such as gparted or clonezilla.  I kept getting "Live environment could not be found on the network".  This is after setting: fetch=https://my.server.com/gparted/filesystem.squashfs

I tried hitting the web server on http directly with the IP address an it worked! woo! So either "Fetch" cannot work with DNS, or something else is going on.

DHCP is clearly working as the booted image gets a valid IP address and can hit one of my web servers.  Even if I add:
ip=dhcp or just dhcp to the kernel paramaters, it doesnt work.  

IPXE itself works fine. I chainload using https and the server name and no issues.  Not sure what else I could do?  My organization rotates AD servers (And thus, DNS servers) so manually setting a DNS server wouldnt be very helpful long term.  Is there something I am missing?

**Comment by NiKiZe** [comment]
Have you checked if your resolv file gets updated correctly?
But I've also read about busybox builds without DNS support, you might be hitting one of those.

Check if you can download the squashfs from iPXE instead and have the distro scripts find and use that. (gentoo.ipxe.se describes and uses that approach)

**Comment by icsy7867** [comment]
Awesome quick response! Perhaps it is that kernel "fetch" command just doesnt use DNS.  I spent hours on that lol.

I will check out the gentoo.ipxe.org! Thank you for the guidance. Do you have a link to the gentoo.ipxe.org file? I am having trouble finding it.

**Reply by NiKiZe** [reply]
The kernel has no fetch command, what happens is that the init scripts parses the kernel cmdline, detects fetch, probably calls wget (the busybox implementation) which tries to do the download.

The important part to understand here is that all of this is Linux userspace, not kernel. And as such is also different between distributions.

**Reply by icsy7867** [reply]
That is all beginning to make more sense!  Interesting, after I boot into a distro using the IP address to get the squashfs file, DNS is working fine.

Do you have a direct link to the ipxe file you mentioned? gentoo.ipxe.org?

Thank you!

**Reply by NiKiZe** [reply]
I do apologize, the correct link is https://gentoo.ipxe.se/

The preferred approach is probably in
https://gentoo.ipxe.se/altcombine.ipxe (I need to find time to clean these up)
Do read the linked bug report where Gentoo was updated to support this way of booting out of the box.

**Reply by icsy7867** [reply]
No need to apologize I appreciate the fast responses and information! I will dig into this.

**Comment by icsy7867** [comment]
So after digging into this more, it's not a DNS issue at all.  It's an https issue.  The fetch command uses wget (I believe), and in the initrd userspace cannot use https.  Whether that is because the CA cert needs to be there, or SSL/TLS capabilities just arent there I am not sure yet.

Since my webserver is behind a reverse proxy and using docker containers, I simply exposed a port directly to my container, port 8888 in this case.

When I did:
fetch=http://my.server.com:8888/path/to/filesystem.squashfs

It worked fine.  So when I get the chance I am going to take a look at the scripts inside of initrd and see if that is something I can modify.  If it is just the https being untrusted, I think I can test adding a:
```
--no-check-certificate
```
to the wget command.
